packages/cyberstorm-forms/src/components/FormSelect.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import { z, ZodObject } from "zod";
+import { ZodRawShape } from "zod/lib/types";
+import { Path, useController } from "react-hook-form";
+import styles from "./FormSelect.module.css";
+import * as Select from "@radix-ui/react-select";
+import React, { PropsWithChildren } from "react";
+import { Icon } from "@thunderstore/cyberstorm";
+import { faCaretDown, faCheckCircle } from "@fortawesome/pro-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export type FormSelectProps<
+  Schema extends ZodObject<Z>,
+  Z extends ZodRawShape
+> = {
+  // The schema is required to allow TS to infer valid values for the name field
+  schema: Schema;
+  name: Path<z.infer<Schema>>;
+  options: { value: string; label: string }[];
+  placeholder?: string;
+  defaultValue?: string;
+  ariaLabel?: string;
+};
+export function FormSelect<Schema extends ZodObject<Z>, Z extends ZodRawShape>({
+  name,
+  defaultValue,
+  options,
+  ariaLabel,
+}: FormSelectProps<Schema, Z>) {
+  const {
+    field,
+    fieldState: { error },
+    formState: { isSubmitting, disabled },
+  } = useController({ name });
+
+  if (defaultValue) {
```

Comment:
I'm not familiar with the whole form contraption we've got going on, but I feel that setting a default value as an effect, field by field, in a possibly large form, is not the approach we want to take. On a quick glance RHF supports setting defaults once per form. If possible, I think that approach should be used. @MythicManiac feel free to chip in.

Reply:
We'll need a way to update the value in the field later on, when we are not reloading the page. But that is then, so I'll remove this.

There was a problem with using the defaultValue solely, which I can't remember now, but I'll figure that one out.